### PR TITLE
chore(mls): unify MLSClientIdentity models (WPB-9774)

### PR DESCRIPTION
**Pull request by mchenani**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
To fetch the MLSClientIdentity from CC we were mapping them in different models, due to different changes we needed more data to be exposed, we needed them to be unified and map all the available data from CC.

### Issues

Hard to maintain different models
Due to maturity of CC now we have a rich and solid model we can map full object in Kalium and Android App.

Needs releases with:
With AR
- [x] GitHub link to other pull request

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

All E2EI features must work as before!

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/client/ObserveE2EIRequiredUseCaseTest.kt [hunk 1]
```diff
@@ -379,5 +382,34 @@ class ObserveE2EIRequiredUseCaseTest {
             thumbprint = "thumbprint",
             endAt = DateTimeUtil.currentInstant().plus(1.days)
         )
+        val CRYPTO_CLIENT_ID = CryptoQualifiedClientId("clientId", TestConversation.USER_1.toCrypto())
+        val VALID_X509Identity = X509Identity(
+//             WireIdentity.Handle(
```
Comment (mchenani) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/client/ObserveE2EIRequiredUseCaseTest.kt:387]: updating this

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/ObserveE2EIRequiredUseCase.kt [hunk 2]
```diff
@@ -76,30 +74,37 @@ internal class ObserveE2EIRequiredUseCaseImpl(
...
-
-                            is GetE2EICertificateUseCaseResult.Success -> onUserHasCertificate(certificateResult.certificate)
-
-                            is GetE2EICertificateUseCaseResult.NotActivated -> onUserHasNoCertificate(setting)
+                    return@map currentClientIdProvider().fold(
+                        {
+                            E2EIRequiredResult.NotRequired
+                        },
+                        { clientId ->
+                            E2EIRequiredResult.NotRequired
```
Comment (borichellow) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/ObserveE2EIRequiredUseCase.kt:82]: this line does nothing

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/ObserveE2EIRequiredUseCase.kt [hunk 3]
```diff
@@ -76,30 +74,37 @@ internal class ObserveE2EIRequiredUseCaseImpl(
...
         } ?: E2EIRequiredResult.NoGracePeriod.Create
 
-    private fun onUserHasCertificate(certificate: E2eiCertificate) =
-        if (certificate.status == CertificateStatus.EXPIRED) {
-            E2EIRequiredResult.NoGracePeriod.Renew
-        } else if (certificate.status == CertificateStatus.VALID && certificate.shouldRenew()) {
-            E2EIRequiredResult.WithGracePeriod.Renew(certificate.renewGracePeriodLeft())
+    private fun onUserHasValidCertificate(identity: X509Identity) =
+        if (identity.shouldRenew()
+        ) {
```
Comment (borichellow) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/ObserveE2EIRequiredUseCase.kt:106]: fix the formatting

**Review by borichellow — APPROVED** [review]
great job! Now it looks much better and understendable :)

**Review by vitorhugods — APPROVED** [review]

**Pull request merged**